lib/thread.c

```diff
@@ -1214,6 +1286,10 @@ void thread_cancel_event(struct thread_master *master, void *arg)
 {
 	assert(master->owner == pthread_self());
 
+	/* Only worth anything if caller supplies an arg. */
+	if (arg == NULL)
```

Comment:
should we log something here?  This seems like a developer has made a mistake of some sort?

Reply:
I could assert I guess? a log seems like something that no one's really going to see - dunno. I was half-thinking that it was ok for it to be benign to call with a NULL - it might just mean that some datastruct had a NULL, and the cleanup code didn't have to test that cell before calling the 'cancel' api?